### ISSUE TITLE
Elm 0.19 does not work natively on iOS

### ISSUE THREAD

**Issue by stormont** [elm/core#1014]
This is partly related to #998.

**Elm 0.19 does not appear to work natively on iOS.** Converting an app to 0.19 and updating the `Elm.Main.init(...)` statement are the sole changes I have made to reproduce this. I'm able to reproduce this issue with both `UIWebView` and `WKWebView`.

There are two issues here:
- Like in #998, the `Elm` variable does not get scoped correctly. The solution described in that issue only partly works (the final invoke must simply be `(window)`, rather than the suggested `(this || window)` to get the variable definition scoped correctly to be accessible externally as `Elm`).
- Once this is resolved, loading Elm with `Elm.Main.init({node: node});` does not work (yes, my module is named `Main`). I'm uncertain where it breaks down at this point; I see no rendering occur. I inserted numerous `Debug.log` statements in my Elm code and saw none of them fire - including right at the `main` entry point. It appears to not be able to get to the point of firing `main` inside the call to `Elm.Main.init(...)`. **There's some scheduler magic going on in this call; could the breakdown be here?**

Where is it _not_ broken?
- Both `UIWebView` and `WKWebView` worked correctly with the same app on 0.18. No changes have been made to the Swift code.
- The exact same Elm code (including HTML loader) works correctly in a Chrome browser on an OS X laptop.
- The exact same Elm code (including HTML loader) works correctly in an Android `WebView`.
- Likewise, I'm able to reproduce the issue with/without `uglifyjs` or with/without the `--optimize` flag during compilation. Neither of these seem to be the issue.
- I can confirm that the Elm code is being loaded by the web view. The global variable (after the invoke fix cited above) is available. I can also confirm that ports appear to be available for subscription. So, as far as I can see, variables appear to scoped correctly.

**Comment by stormont** [comment]
Closing to move to elm/compiler

**Comment by supermario** [comment]
@stormont where did you get to with this issue? I can't seem to find any issues in elm/compilier where you might have taken it – are you able to link it please? 😊

**Comment by stormont** [comment]
After further digging, I think there's a conflict with how the Elm plugin for Visual Studio Code runs background compilation when a file is saved. I was able to reproduce the problem outside of iOS rendering, but I haven't been able to reliably able to generate a true reproduce case to pin it down. At this point, I'm not sure whether the problem is in the 0.19 compiler or something to do with the VS Elm plugin, so I haven't reopened this (I'm not sure which repo to log it against right now).

In the meantime, I've been able to get around it by always fully recompiling at the command line (`rm -rf elm-stuff/ && elm make ...`). Given how fast the compiler is now, it's not causing me enough pain to dig deeper at the moment.
